src/test/java/uk/gov/companieshouse/company_appointments/ResourceChangedApiServiceTest.java

```diff
@@ -0,0 +1,151 @@
+
+    @Mock
+    private ChangedResource changedResource;
+
+    @InjectMocks
+    private ResourceChangedApiService resourceChangedApiService;
+
+    @Test
+    @DisplayName("Test should successfully invoke chs-kafka-api")
+    void invokeChsKafkaApi() throws ApiErrorResponseException, ServiceUnavailableException {
+        // given
+        when(apiClientService.getInternalApiClient()).thenReturn(internalApiClient);
+        when(internalApiClient.privateChangedResourceHandler()).thenReturn(privateChangedResourceHandler);
+        when(privateChangedResourceHandler.postChangedResource(any(), any())).thenReturn(changedResourcePost);
+        when(changedResourcePost.execute()).thenReturn(response);
+        when(mapper.mapChangedResource(resourceChangedRequest)).thenReturn(changedResource);
+
+        // when
+        resourceChangedApiService.invokeChsKafkaApi(resourceChangedRequest);
+
+        // then
+        verify(apiClientService).getInternalApiClient();
+        verify(internalApiClient).privateChangedResourceHandler();
+        verify(privateChangedResourceHandler).postChangedResource("/resource-changed", changedResource);
+        verify(changedResourcePost).execute();
+    }
+
+    @ParameterizedTest
+    @MethodSource("invokeChsKafkaApiExceptionFixtures")
+    void invokeChsKafkaApiExceptionTests(ResourceChangedApiServiceTestArgument argument) throws ApiErrorResponseException {
+        // given
+        HttpResponseException.Builder builder = new HttpResponseException.Builder(argument.getStatusCode(), argument.getErrorMessage(), new HttpHeaders());
+        ApiErrorResponseException apiErrorResponseException = new ApiErrorResponseException(builder);
+
+        when(apiClientService.getInternalApiClient()).thenReturn(internalApiClient);
+        when(internalApiClient.privateChangedResourceHandler()).thenReturn(privateChangedResourceHandler);
+        when(privateChangedResourceHandler.postChangedResource(any(), any())).thenReturn(changedResourcePost);
+        when(mapper.mapChangedResource(resourceChangedRequest)).thenReturn(changedResource);
+        when(changedResourcePost.execute()).thenThrow(apiErrorResponseException);
+
+        // when
+        Executable executable = () -> resourceChangedApiService.invokeChsKafkaApi(resourceChangedRequest);
+
+        // then
+        assertThrows(ServiceUnavailableException.class, executable);
+        verify(apiClientService, times(1)).getInternalApiClient();
+        verify(internalApiClient, times(1)).privateChangedResourceHandler();
+        verify(privateChangedResourceHandler, times(1)).postChangedResource("/resource-changed", changedResource);
+        verify(changedResourcePost, times(1)).execute();
+    }
+
+    private static Stream<Arguments> invokeChsKafkaApiExceptionFixtures() {
+        return Stream.of(
+                Arguments.of(
+                        Named.of("Throws service unavailable exception when response code is HTTP 500",
```

Comment:
The name says 500, but the exception uses a code of 503. 
See the next element where the numbers are swapped.
Should the text and codes agree?

Reply:
Addressed in latest commit